File: examples/simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/CustomApplication.java
package com.zhuinden.simplestackexamplemvvm.application;

import android.app.Application;

import com.zhuinden.simplestackexamplemvvm.application.injection.AndroidModule;
import com.zhuinden.simplestackexamplemvvm.application.injection.ApplicationComponent;
import com.zhuinden.simplestackexamplemvvm.application.injection.DaggerApplicationComponent;

/**
 * Created by Zhuinden on 2017.07.26..
 */

public class CustomApplication
        extends Application {
    private static CustomApplication INSTANCE;

    ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        INSTANCE = this;
        applicationComponent = DaggerApplicationComponent.builder().androidModule(new AndroidModule(this)).build();
    }

    public static CustomApplication get() {
        return INSTANCE;
    }

    public ApplicationComponent appComponent() {
        return applicationComponent;
    }
}

File: samples/legacy-architecture-samples/simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/CustomApplication.kt
package com.zhuinden.simplestackexamplemvvm.application


import android.app.Application
import com.zhuinden.simplestack.GlobalServices
import com.zhuinden.simplestackexamplemvvm.application.injection.DaggerApplicationComponent
import com.zhuinden.simplestackextensions.servicesktx.add

/**
 * Created by Zhuinden on 2017.07.26..
 */
class CustomApplication : Application() {
    lateinit var globalServices: GlobalServices
        private set

    override fun onCreate() {
        super.onCreate()

        val applicationComponent = DaggerApplicationComponent.factory().create(this)

        globalServices = GlobalServices.builder()
            .add(applicationComponent)
            .build()
    }
}